common/numerated/src/tree.rs

```diff
@@ -203,6 +200,7 @@ impl<T: Numerated> IntervalsTree<T> {
 
         if let Some(right_end) = right_end.inc_if_lt(start) {
```

Comment:
comment all of the cases pls for whole algo with points alignment in docs like `start < right_start <= right_end < end`